flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/AbstractHandler.java

```diff
@@ -242,8 +242,14 @@ private CompletableFuture<Void> handleException(
             return CompletableFuture.completedFuture(null);
         }
         int maxLength = flinkHttpObjectAggregator.maxContentLength() - OTHER_RESP_PAYLOAD_OVERHEAD;
-        if (throwable instanceof RestHandlerException) {
-            RestHandlerException rhe = (RestHandlerException) throwable;
+        if (throwable instanceof RestHandlerException
+                || throwable.getCause() instanceof RestHandlerException) {
```

Comment:
That's a strange change (checking for both, the exception and the exception's cause), don't you think? It might be an indication that we should do the change somewhere else. WDYT? :thinking:

Reply:
Yes, I can throw `RestHandlerException `directly instead as a cause of `CompletionException`.